src/components/pages/List/List.js

```diff
@@ -0,0 +1,40 @@
+import React, { useState, useEffect } from 'react';
+import { navigate } from '@reach/router';
+import Button from 'react-bootstrap/Button';
+
+import Search from '../../Search';
+import OfferList from '../../OfferList';
+import '../styles.scss';
+
+function List({ categoryID }) {
+  const [categoryName, setCategoryName] = useState('');
+  const [offers, setOffers] = useState([]);
+
+  useEffect(() => {
+    fetch(`/api/categories/${categoryID}`)
+      .then(response => response.json())
+      .then(data => setCategoryName(data.name));
+
+    fetch(`/api/categories/${categoryID}/offers`)
```

Comment:
Taka można by złapać `.catch` (przykładowo w przypadku jeśli użytkownik przejdzie pod URL z ID kategorii, która nie istnieje) i wtedy przekierowywać pod `404, not found`

Nie mówię, że już tutaj trzeba to zrobić, ale to potencjalny dodatek, który można zaimplementować później.